src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -479,15 +474,23 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
                 return new ExtendedErrorTypeSymbol(this.Compilation.Assembly.GlobalNamespace, LookupResultKind.NotCreatable, diagnostics.Add(ErrorCode.ERR_TupleTooFewElements, syntax.Location));
             }
 
-            return TupleTypeSymbol.Create(syntax.Location,
-                                            typesArray,
-                                            locationsArray,
-                                            elementNames == null ?
-                                                default(ImmutableArray<string>) :
-                                                elementNames.ToImmutableAndFree(),
-                                            this.Compilation,
-                                            syntax,
-                                            diagnostics);
+            var tupleTypeSymbol = TupleTypeSymbol.Create(
+                syntax.Location,
+                typesArray,
+                locationsArray,
+                elementNames == null ?
+                    default(ImmutableArray<string>) :
+                    elementNames.ToImmutableAndFree(),
+                this.Compilation,
+                syntax,
+                diagnostics);
+
+            if (ShouldCheckConstraints)
```

Comment:
>if (ShouldCheckConstraints) [](start = 12, length = 27)

Do we have a test covering scenario when ShouldCheckConstraints is false here and the caller handles the check?